BGL/include/CGAL/boost/graph/named_params_helper.h

```diff
@@ -112,8 +113,82 @@ namespace CGAL {
     typedef typename boost::property_traits<PMap>::value_type type;
   };
 
+  namespace internal_np
+  {
+  //overloads used to select a default map:
+  // use the one passed in the named parameters (user must have initialized it)
+  template <class MapFromNP, class Default_tag, class Dynamic_tag, class Mesh>
+  MapFromNP
+  get_map(MapFromNP m, Default_tag, Dynamic_tag, const Mesh&, bool& need_init)
+  {
+    need_init = false;
+    return m;
+  }
+
+  // use the one internal to the mesh (user must have initialized it)
+  template <class Default_tag, class Dynamic_tag, class Mesh>
+  typename boost::property_map<Mesh, Default_tag >::const_type
+  get_map(CGAL::internal_np::Param_not_found, Default_tag t, Dynamic_tag , const Mesh& m, bool& need_init)
+  {
+    need_init = false;
+    return get(t,m);
+  }
+
+  // create a dynamic property and initialize it
+  template <class Dynamic_tag, class Mesh>
+  typename boost::property_map<Mesh, Dynamic_tag >::const_type
+  get_map(CGAL::internal_np::Param_not_found, Dynamic_tag t, Dynamic_tag , const Mesh& m, bool& need_init)
+  {
+    need_init = true;
```

Comment:
I think another case when init is needed is when the index map is writable (like Polyhedron with items with ids).